airbyte-integrations/connectors/destination-postgres/src/main/kotlin/io/airbyte/integrations/destination/postgres/typing_deduping/PostgresGenerationIdMigrator.kt

```diff
@@ -0,0 +1,121 @@
+            } else {
+                logger.info(
+                    "Executing migration for ${stream.id.rawNamespace}.${stream.id.rawName} for real"
+                )
+
+                destinationHandler.execute(
+                    Sql.of(
+                        DSL.alterTable(DSL.name(stream.id.rawNamespace, stream.id.rawName))
+                            .addColumn(
+                                DSL.name(JavaBaseConstants.COLUMN_NAME_AB_GENERATION_ID),
+                                SQLDataType.BIGINT.nullable(true)
+                            )
+                            .getSQL(ParamType.INLINED)
+                    )
+                )
+            }
+        }
+
+        val maybeExistingFinalTable =
+            JdbcDestinationHandler.findExistingTable(
+                database,
+                databaseName,
+                stream.id.finalNamespace,
+                stream.id.finalName
+            )
+        if (maybeExistingFinalTable.isEmpty) {
+            logger.info(
+                "Stopping migration for ${stream.id.originalNamespace}.${stream.id.originalName} because the public table doesn't exist"
+            )
+            return Migration.MigrationResult(
+                state.destinationState.copy(isAirbyteGenerationIdPresent = true),
+                false
+            )
+        }
+        val existingFinalTable = maybeExistingFinalTable.get()
+        if (existingFinalTable.columns[JavaBaseConstants.COLUMN_NAME_AB_GENERATION_ID] != null) {
+            // The raw table already has the _airbyte_meta column. No migration necessary. Update
+            // the state.
+            logger.info(
+                "Skipping migration for ${stream.id.finalNamespace}.${stream.id.finalName} because the generation_id column is already present"
+            )
+        } else {
+            logger.info(
+                "Executing migration for ${stream.id.finalNamespace}.${stream.id.finalName} for real"
+            )
+
+            destinationHandler.execute(
+                Sql.of(
+                    DSL.alterTable(DSL.name(stream.id.finalNamespace, stream.id.finalName))
+                        .addColumn(
+                            DSL.name(JavaBaseConstants.COLUMN_NAME_AB_GENERATION_ID),
+                            SQLDataType.BIGINT.nullable(true)
+                        )
+                        .getSQL(ParamType.INLINED)
+                )
+            )
+        }
+
+        // Update the state. We didn't modify the table in a relevant way, so don't invalidate the
+        // InitialState.
+        // We will not do a soft reset since it could be time-consuming, instead we leave the old
+        // data i.e. `errors` instead of `changes` as is since this column is controlled by us.
+        return Migration.MigrationResult(
+            state.destinationState.copy(
+                needsSoftReset = false,
+                isAirbyteGenerationIdPresent = true
+            ),
+            false
```

Comment:
I think we _should_ invalidate the initial state if we altered the final table? Adding a column changes the `isSchemaMismatch` value, so we do need to refetch in that case